cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2920,7 +2926,7 @@ private boolean isPodToRestart(StatefulSet sts, Pod pod, boolean isAncillaryCmCh
                         reasons.add("Pod has old " + ca + " certificate generation");
                     }
                 }
-                if (isAncillaryCmChange) {
+                if (cmAnno != ancillaryCmGeneration) {
```

Comment:
In theory a NPE might be possible here if the `ancillaryCmGeneration` is null. Probably best to do the roll if it's null.